pkg/knowledge_base2/property_types.go

```diff
@@ -545,3 +547,101 @@ func (r *ResourcePropertyType) ZeroValue() any {
 func (p *PropertyRefPropertyType) ZeroValue() any {
 	return construct.PropertyRef{}
 }
+
+func (m *MapPropertyType) Contains(value any, contains any) bool {
```

Comment:
This is more of a subset than contains. Needs documentation too given that `func(any, any)` doesn't really give any signature-related hints to what should be passed in or valid types.